File: court-kiosk/api/health.js
// Vercel serverless function for health check
export default async function handler(req, res) {
  // Set CORS headers
  res.setHeader('Access-Control-Allow-Origin', '*');
  res.setHeader('Access-Control-Allow-Methods', 'GET, OPTIONS');
  res.setHeader('Access-Control-Allow-Headers', 'Content-Type');

  // Handle preflight requests
  if (req.method === 'OPTIONS') {
    return res.status(200).end();
  }

  if (req.method !== 'GET') {
    return res.status(405).json({ error: 'Method not allowed' });
  }

  try {
    return res.status(200).json({
      status: 'OK',
      timestamp: new Date().toISOString(),
      service: 'Court Kiosk API',
      version: '1.0.0'
    });

  } catch (error) {
    console.error('Health check error:', error);
    return res.status(500).json({
      status: 'ERROR',
      error: error.message
    });
  }
}

File: court-kiosk/frontend/api/health.py
from flask import Flask, jsonify
import os

app = Flask(__name__)

@app.route('/api/health', methods=['GET'])
def health_check():
    """Health check endpoint for Vercel deployment"""
    return jsonify({
        "status": "OK",
        "message": "Court Kiosk API is running",
        "environment": os.getenv('VERCEL_ENV', 'development'),
        "version": "1.0.0"
    })

# Vercel serverless function handler
def handler(request):
    with app.test_request_context(request.url, method=request.method, data=request.get_data()):
        return app.full_dispatch_request()
